### DISCUSSION TITLE
Is it possible to use gtk themes on my application?

### DISCUSSION THREAD

**Discussion by Raj2032** [fltk-rs/fltk-rs#1604] [Q&A]
I am using Linux and I am using gtk themes for my DE, was wondering if its possible to get my application to use the GTK theme that is currently set on my DE?

Edit:

```rust
let app = app::App::default().with_scheme(app::Scheme::Gtk);
```

I have added this in, the buttons have changed but the entire background still looks the same, I know you can change the background color manually but see if I changed my DE's default theme, all of my applications get affected but my application still looks the same.

**Comment by MoAlyousef** [comment]
You can try calling `win.show_with_env_args()` or using `app::get_system_colors()` and see if those help. Unfortunately on Linux each distro appears to store its themes in a certain fashion which FLTK might not be able to pick up.

**Reply by Raj2032** [reply]
Unfortunately `win.show_with_env_args()` will hide all the widgets and `app::get_system_colors()` does nothing.

> Unfortunately on Linux each distro appears to store its themes in a certain fashion which FLTK might not be able to pick up.

I am using arch based distro and using xfce. Should I copy themes into the home directory?

**Reply by MoAlyousef** [reply]
That wouldn't work. For example, FLTK on X11 systems use XGetDefault and XParseColor to determine the background, foreground and other system colors. So unless these are set by the desktop environment, FLTK will use its default color theme.
You can try crates like [rusty_themy](https://docs.rs/rusty_themy/latest/rusty_themy/index.html) which might help.

**Reply by Raj2032** [reply]
> That wouldn't work. For example, FLTK on X11 systems use XGetDefault and XParseColor to determine the background, foreground and other system colors. So unless these are set by the desktop environment, FLTK will use its default color theme. You can try crates like [rusty_themy](https://docs.rs/rusty_themy/latest/rusty_themy/index.html) which might help.

Right I see thank you will try this then.
